### PR TITLE
languages: add templ

### PR DESCRIPTION
**Pull request by rsteube**
Added [templ](https://github.com/a-h/templ).

The captures probably still need some renaming.
Is the `structure.scm` needed?

[Quickstart](https://templ.guide/quick-start/creating-a-simple-templ-component)
[IDE Support](https://templ.guide/commands-and-tools/ide-support)

### CODE REVIEW

File: runtime/queries/templ/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  (interpreted_string_literal)
+  (raw_string_literal)
+  (rune_literal)
+] @string
+
+; Comments
+
+(comment) @comment @spell
+
+(element_comment) @comment @spell
```
Comment (the-mikedavis) [inline runtime/queries/templ/highlights.scm:105]:
```suggestion
(comment) @comment

(element_comment) @comment
```

File: runtime/queries/templ/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+; * https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/go/highlights.scm
+; * https://github.com/tree-sitter/tree-sitter-go/blob/master/queries/highlights.scm
+
+(package_identifier) @namespace
+
+(parameter_declaration (identifier) @variable.parameter)
+(variadic_parameter_declaration (identifier) @variable.parameter)
+
+(function_declaration
+  name: (identifier) @definition.function)
```
Comment (the-mikedavis) [inline runtime/queries/templ/highlights.scm:13]:
```suggestion
  name: (identifier) @function)
```

File: runtime/queries/templ/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+(type_spec name: (type_identifier) @type.definition)
```
Comment (the-mikedavis) [inline runtime/queries/templ/highlights.scm:15]:
```suggestion
(type_spec name: (type_identifier) @type)
```

File: runtime/queries/templ/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+(type_identifier) @type
+(field_identifier) @property
```
Comment (the-mikedavis) [inline runtime/queries/templ/highlights.scm:17]:
```suggestion
(field_identifier) @variable.other.member
```

File: runtime/queries/templ/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+(identifier) @variable
+
+; Function calls
+
+(call_expression
+  function: (identifier) @reference.call)
+
+(call_expression
+  function: (selector_expression
+    field: (field_identifier) @reference.call))
```
Comment (the-mikedavis) [inline runtime/queries/templ/highlights.scm:27]:
```suggestion
  function: (identifier) @function)

(call_expression
  function: (selector_expression
    field: (field_identifier) @function))
```

File: runtime/queries/templ/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  "func"
+  "var"
+  "const"
+  "templ"
+  "css"
+  "type"
+  "struct"
+  "range"
+  "script"
+] @keyword.storate.type
```
Comment (the-mikedavis) [inline runtime/queries/templ/highlights.scm:74]:
```suggestion
] @keyword.storage.type
```

File: runtime/queries/templ/structure.scm
Comment (the-mikedavis) [inline runtime/queries/templ/structure.scm:1]: This can be removed

**Review by the-mikedavis — CHANGES_REQUESTED** [review]

**Comment by the-mikedavis** [comment]
It looks like the tree-sitter-templ scanner.c causes has some compilation errors on Windows

**Comment by rsteube** [comment]
```c
  D:\a\helix\helix\runtime\grammars\sources\templ\src\scanner.c(67): warning C4189: 'found': local variable is initialized but not referenced
  D:\a\helix\helix\runtime\grammars\sources\templ\src\scanner.c(93): error C2016: C requires that a struct or union have at least one member
  D:\a\helix\helix\runtime\grammars\sources\templ\src\scanner.c(341): warning C4244: 'function': conversion from 'int32_t' to 'wint_t', possible loss of data
```

Is it related to https://github.com/vrischmann/tree-sitter-templ/blob/master/src/scanner.c#L93 not having any member?
Or is that error just a warning and something other causes the failed build?

**Comment by the-mikedavis** [comment]
Yep, I believe the warnings are benign but the error from the line you linked stops compilation

File: languages.toml [hunk 7]
```diff
@@ -2998,3 +2999,17 @@ roots = []
...
+
+[[language]]
+name = "templ"
+scope = "source.templ"
+file-types = ["templ"]
+roots = ["go.work", "go.mod"]
+comment-token = "//"
+indent = { tab-width = 2, unit = "  " }
+language-servers = [ "templ" ]
+grammar = "templ"
```
Comment (the-mikedavis) [inline languages.toml:3011]: This key can be eliminated - when it is omitted the name of the language is used by default

**Review by the-mikedavis — COMMENTED** [review]
Just a minor note in the languages.toml, otherwise I think this looks good

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**